arithmetization/src/main/java/net/consensys/linea/zktracer/module/oob/OobChunk.java

```diff
@@ -0,0 +1,1294 @@
+      isRipemd = true;
+      wghtSum = Bytes.fromHexString("FF03").toUnsignedBigInteger();
+    } else if (target.equals(Address.ID)) {
+      isIdentity = true;
+      wghtSum = Bytes.fromHexString("FF04").toUnsignedBigInteger();
+    } else if (target.equals(Address.ALTBN128_ADD)) {
+      isEcadd = true;
+      wghtSum = Bytes.fromHexString("FF06").toUnsignedBigInteger();
+    } else if (target.equals(Address.ALTBN128_MUL)) {
+      isEcmul = true;
+      wghtSum = Bytes.fromHexString("FF07").toUnsignedBigInteger();
+    } else if (target.equals(Address.ALTBN128_PAIRING)) {
+      isEcpairing = true;
+      wghtSum = Bytes.fromHexString("FF08").toUnsignedBigInteger();
+    } else if (target.equals(Address.BLAKE2B_F_COMPRESSION)) {
+      if (blake2FCallNumber == 1) {
+        isBlake2FCds = true;
+        wghtSum = Bytes.fromHexString("FA09").toUnsignedBigInteger();
+      } else if (blake2FCallNumber == 2) {
+        isBlake2FParams = true;
+        wghtSum = Bytes.fromHexString("FB09").toUnsignedBigInteger();
+      }
+    } else if (target.equals(Address.MODEXP)) {
+      switch (modexpCallNumber) {
+        case 1:
+          isModexpCds = true;
+          wghtSum = Bytes.fromHexString("FA05").toUnsignedBigInteger();
+        case 2:
+          isModexpXbs = true;
+          isModexpBbs = true;
+          wghtSum = Bytes.fromHexString("FB05").toUnsignedBigInteger();
+        case 3:
+          isModexpXbs = true;
+          isModexpEbs = true;
+          wghtSum = Bytes.fromHexString("FB05").toUnsignedBigInteger();
+        case 4:
+          isModexpXbs = true;
+          isModexpMbs = true;
+          wghtSum = Bytes.fromHexString("FB05").toUnsignedBigInteger();
+        case 5:
+          isModexpLead = true;
+          wghtSum = Bytes.fromHexString("FC05").toUnsignedBigInteger();
+        case 6:
+          prcModexpPricing = true;
+          wghtSum = Bytes.fromHexString("FD05").toUnsignedBigInteger();
+        case 7:
+          prcModexpExtract = true;
+          wghtSum = Bytes.fromHexString("FE05").toUnsignedBigInteger();
+      }
+    } else {
+      throw new IllegalArgumentException("Precompile not relevant for Oob");
+    }
+    oobInst = wghtSum;
+  }
+
+  public boolean isInst() {
+    return isJump
+        || isJumpi
+        || isRdc
+        || isCdl
+        || isCall
+        || isXCall
+        || isCreate
+        || isSstore
+        || isDeployment;
+  }
+
+  public boolean isCommonPrecompile() {
+    return isEcRecover || isSha2 || isRipemd || isIdentity || isEcadd || isEcmul || isEcpairing;
+  }
+
+  public boolean isBlakePrecompile() {
+    return isBlake2FCds || isBlake2FParams;
+  }
+
+  public boolean isModexpPrecompile() {
+    return isModexpCds || isModexpXbs || isModexpLead || prcModexpPricing || prcModexpExtract;
+  }
+
+  public boolean isPrecompile() {
+    return isCommonPrecompile() || isBlakePrecompile() || isModexpPrecompile();
+  }
+
+  public int maxCt() {
```

Comment:
prefer a return switch

Reply:
oh the boolean flag are not exclusive ?

Reply:
They are, but I do not get what is the advantage of using a switch case. Do you believe it would be more readable? How would you write it?